hw4/04-templates.js

```diff
@@ -1,61 +1,99 @@
-const express = require('express');
-const axios = require('axios');
+const express = require("express");
+const fetch = require("node-fetch");
 
 const app = express();
 const port = process.env.PORT || 5001;
 
 // Use Pug as the templating engine
-app.set('views', __dirname + '/views');
-app.set('view engine', 'pug');
+app.set("views", __dirname + "/views");
+app.set("view engine", "pug");
 
 // REST Countries URL
-const url = 'https://restcountries.com/v3.1/all';
+const url = "https://restcountries.com/v3.1/all";
 
-// Add your code here
+let data = [];
+let countries = [], populous = [], regions = [];
+let isPopulationDataCalculated = false;
+let isRegionDataCalculated = false;
+let regionObject = {};
 
-app.get('/', (req, res) => {
+app.get("/", async (req, res) => {
   // render pug template for the index.html file
 
-  res.render('index', {
-    heading: 'Countries of the World',
-    main: 'Welcome to this application. Using the REST Countries API, we will be showing the countries and capitals of the world, the most populous countries in the world, and the number of countries in each region of the world',
+  data = await fetch(url)
+    .then(data => data.json())
+    .then(data => data)
+    .catch(e => {
+        console.log(e);
+        return [];
+      }
+    );
+
+  res.render("index", {
+    heading: "Countries of the World",
+    main: "Welcome to this application. Using the REST Countries API, we will be showing the countries and capitals of the world, the most populous countries in the world, and the number of countries in each region of the world"
   });
 });
 
-app.get('/capitals', (req, res) => {
+app.get("/capitals", (req, res) => {
   // map the output array to create an array with country names and capitals
   // check for empty data in the output array
-
-  let countries = ['Afghanistan', 'Aland Islands', 'Albania'];
-
-  res.render('page', {
-    heading: 'Countries and Capitals',
-    results: countries,
+  if (countries.length === 0) {
+    countries = data
+      .map((country, i) => `${country.name.common}  ${country.capital && country.capital[0] ? (`- ${country.capital[0]}`) : ""}`)
```

Comment:
Per screenshots on Canvas you should probably check if the capital exists for the country because certain countries like Antartica or Macau do not have capitals

Reply:
yup, I'm checking for the capital. Regardless of capital, I'm printing country name.